src/cli/build.cpp

```diff
@@ -0,0 +1,136 @@
+// vim: ts=4 sts=4 sw=4 et
+#include <string>
+
+#include <fmt/core.h>
+#include <fmt/ranges.h>
+#include <fmt/std.h>
+#include <nlohmann/json.hpp>
+#include <spdlog/spdlog.h>
+
+#include <filesystem>
+#include <util/subprocess.h>
+
+#include "build.h"
+#include "cli/terminal.h"
+#include "fmt/base.h"
+#include "help.h"
+#include "site/site.h"
+#include "util/curl.h"
+#include "util/fs.h"
+
+namespace uenv {
+
+std::string build_footer();
+
+std::string format_reply(const std::string&);
+
+void build_args::add_cli(CLI::App& cli,
+                         [[maybe_unused]] global_settings& settings) {
+    auto* build_cli =
+        cli.add_subcommand("build", "build a uenv from a local recipe");
+    build_cli->add_flag("-d,--develop", spack_develop, "Assume spack@develop");
+    build_cli->add_option("-n,--name", name, "Name")->required(true);
```

Comment:
For consistency with the other commands in the interface, try using a positional argument of the form
```
name/version:tag@system%uarch
```
e.g.
```
uenv build $recipe_path sirius/2.3@daint%gh200
```

In this case, we let the build service set the tag, so it can be omitted.

Then you have a single argument to parse, see:
https://github.com/eth-cscs/uenv2/blob/main/src/cli/add_remove.cpp#L30

Reply:
I've changed it accordingly.